### PR TITLE
Added Boss and implemented in tests

### PR DESCRIPTION
**Pull request by jackmounts**
first implementation of a simple Boss

### CODE REVIEW

File: src/main/java/eu/eutampieri/catacombs/model/Boss.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
+package eu.eutampieri.catacombs.model;
+
+import eu.eutampieri.catacombs.model.map.TileMap;
+
+public final class Boss extends Entity {
```
Comment (eutampieri) [inline src/main/java/eu/eutampieri/catacombs/model/Boss.java:5]: `implements LivingCharacter` viene fuori automaticamente da `Entity`?

File: src/test/java/eu/eutampieri/catacombs/tests/CharactersTests.java [hunk 2]
```diff
@@ -86,4 +90,10 @@ void testBatUpdate() {
         BAT.update(10);
         // TODO implement checks
     }
+
+    @Test
+    void testBossUpdate() {
+        BOSS.update(10);
+        // TODO implement checks
+    }
```
Comment (eutampieri) [inline src/test/java/eu/eutampieri/catacombs/tests/CharactersTests.java:98]: Not testing much, is the codecov still ok?

**Review by Roccubalbetta — APPROVED** [review]

File: src/main/java/eu/eutampieri/catacombs/model/Boss.java (hunk 1 above)
Comment (jackmounts) [reply src/main/java/eu/eutampieri/catacombs/model/Boss.java:5]: Entity extends GameObject implements LivingCharacter

File: src/test/java/eu/eutampieri/catacombs/tests/CharactersTests.java (hunk 2 above)
Comment (jackmounts) [reply src/test/java/eu/eutampieri/catacombs/tests/CharactersTests.java:98]: cod cov passed...

**Pull request merged**